### PR TITLE
Iterating on Web guide

### PR DESCRIPTION
**Pull request by nbarbettini**
A few changes to shared text:

1. Changed Middleware -> Package, more general terminology that better applies to different stacks.
2. Rename "Define a Callback Route" to "Understand the Callback Route" - there is no code to write here, it is really just a step to help the reader understand what the callback route is _for_.
3. Uncommented the link to Customize Hosted Sign-in Page guide 🎉 
4. Moved into shared text the fact that you can define protected areas of the application that always require authentication. This was previously called out in each stack snippet so it made sense to factor out. This means that the stack snippets are _only_ about building a sign-in link and nothing else (helps it flow better).

A few changes to .NET code examples:
* Removed Bootstrap- and other library-specific code from example code (KISS)
* Reacted to 3 above

### CODE REVIEW

**Comment by nbarbettini** [comment]
And don't worry @jakubvul, I will fix this up once the Great Migration changes all the URLs.

File: packages/@okta/vuepress-site/guides/sign-into-web-app/require-authentication/aspnet/reqauthspecific.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Use the `Authorize` attribute on controllers or actions to require a logged-in user:
+Use the `[Authorize]` attribute on controllers or actions to require a logged-in user:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/guides/sign-into-web-app/require-authentication/aspnet/reqauthspecific.md]: just noticed this so thought I'd add a note:

Use the `[Authorize]` attribute on controllers or actions to require a signed-in user:

File: packages/@okta/vuepress-site/guides/sign-into-web-app/require-authentication/aspnetcore/reqauthspecific.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-Use the `Authorize` attribute on controllers or actions to require a logged-in user:
+Use the `[Authorize]` attribute on controllers or actions to require a logged-in user:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/guides/sign-into-web-app/require-authentication/aspnetcore/reqauthspecific.md]: same

Use the `[Authorize]` attribute on controllers or actions to require a signed-in user:

File: packages/@okta/vuepress-site/guides/sign-into-web-app/require-authentication/aspnet/reqauthspecific.md (hunk 1 above)
Comment (nbarbettini) [reply packages/@okta/vuepress-site/guides/sign-into-web-app/require-authentication/aspnet/reqauthspecific.md]: Good catch 👍

**Comment by nbarbettini** [comment]
Responded to feedback and updated post-migration. Shipping after tests pass.

**Comment by nbarbettini** [comment]
cc @bdemers

**Pull request merged**